### PR TITLE
Intensity p rays

### PR DESCRIPTION
**Pull request by Rodot-**
Add the intensity p-ray values (as well as the p-ray impact parameters) to the runner.  

**Description**

The intensity p-rays are indexed by frequency then p.  Values are in cgs (the array itself is unitless).  Can be accessed by runner.I_nu_p and runner.p_rays after running the integrated spectrum.

**Motivation and context**
Requested

**How has this been tested?**
- [x] Testing pipeline.
- [x] Other. 
Currently an assertion just happens inside the formal integral to make sure the integrated intensity adds up to the computed luminosity.

**Examples**

**Type of change**

- [ ] Bug fix. 
- [x] New feature. 
- [ ] Breaking change. 
- [ ] None of the above. 

**Checklist**

- [x] My change requires a change to the documentation.
    - [ ] I have updated the documentation accordingly.
    - [ ] (optional) I have built the documentation on my fork following [the instructions](https://tardis-sn.github.io/tardis/contributing/development/documentation_guidelines.html#sharing-the-built-documentation-in-your-pr-documentation-preview).
- [x] I have assigned and requested two reviewers for this pull request.

### CODE REVIEW

File: tardis/montecarlo/montecarlo_numba/formal_integral.py [hunk 1]
```diff
@@ -565,6 +567,15 @@ def formal_integral(self, nu, N):
...
         )
+        R_max = self.runner.r_outer_i[-1]
+        ps = calculate_p_values(R_max, N)[None, :]
+        I_nu_p[:, 1:] /= ps[:, 1:]
+        self.runner.I_nu_p = I_nu_p
+        self.runner.p_rays = ps
+
+        I_nu = self.runner.I_nu_p*ps
+        L_test = np.array([8 * M_PI * M_PI * trapezoid_integration((I_nu)[i, :], R_max / N) for i in range(nu.shape[0])])
+        assert np.max(np.abs((L_test-L)/L)) < 1e-7, "Incorrect I_nu_p values, max relative difference:{}".format(np.max(np.abs((L_test-L)/L)))
```
Comment (andrewfullard) [inline tardis/montecarlo/montecarlo_numba/formal_integral.py:578]: Make an issue to turn this into a test and link this PR

**Review by andrewfullard — CHANGES_REQUESTED** [review]

File: tardis/montecarlo/montecarlo_numba/formal_integral_cuda.py [hunk 2]
```diff
@@ -282,8 +282,9 @@ def formal_integral(
             z,
             shell_id,
         )
+        
 
-        return L
+        return L, I_nu
```
[the remark this answers is not in the document]
Comment (Rodot-) [reply tardis/montecarlo/montecarlo_numba/formal_integral_cuda.py:287]: The assert does work for both versions.  It just a matter of the variable names chosen between the two implementations to minimize code changes

File: tardis/montecarlo/montecarlo_numba/tests/test_cuda_formal_integral.py [hunk 3]
```diff
@@ -375,7 +375,7 @@ def test_full_formal_integral(
         formal_integrator_cuda.runner.tau_sobolevs_integ,
         formal_integrator_cuda.runner.electron_densities_integ,
         formal_integrator_cuda.points,
-    )
+    )[0]
```
[the remark this answers is not in the document]
Comment (Rodot-) [reply tardis/montecarlo/montecarlo_numba/tests/test_cuda_formal_integral.py:378]: That's fair, but these tests will be rewritten pretty soon anyway

File: tardis/montecarlo/montecarlo_numba/formal_integral.py (hunk 1 above)
Comment (Rodot-) [reply tardis/montecarlo/montecarlo_numba/formal_integral.py:584]: Done

**Review by andrewfullard — APPROVED** [review]

**Pull request merged**